test/blackbox-tests/test-cases/pkg/portable-lockdirs/portable-lockdirs-package-variable-typo-detection.t

```diff
@@ -0,0 +1,41 @@
+Detect common typos with package variables when describing platforms to solve for.
+
+  $ . ../helpers.sh
+  $ mkrepo
+  $ add_mock_repo_if_needed
+
+Create a workspace with some typos in package variable names
+  $ cat > dune-workspace <<EOF
+  > (lang dune 3.18)
+  > (repository
+  >  (name mock)
+  >  (url "file://$(pwd)/mock-opam-repository"))
+  > (lock_dir
+  >  (repositories mock)
+  >  (solve_for_platforms
+  >   ((arch x86_64)
+  >    (os linux)
+  >    (os_distribution debian)
+  >    (os_family debian))))
+  > EOF
```

Comment:
Sidenote on this: I think it would be useful to add environment variables that force Dune to assume a specific OS, distribution etc., otherwise any lockdir tests will be very dependent on which OS the test is being run.

Reply:
Good idea! I'll do that in a new PR as it doesn't really affect this change.

Reply:
https://github.com/ocaml/dune/pull/11864